File: client/src/modules/exercises/models/QuestionWriteRequest.ts
import type { QuestionPlacement } from '@/types/QuestionPlacement'
import type { QuestionType } from '@/types/QuestionType'

import type { QuestionOptionWriteRequest } from './QuestionOptionWriteRequest'

export interface QuestionWriteRequest {
  text: string
  type: QuestionType
  placement?: QuestionPlacement
  videoTimestampSeconds?: number | null
  explanation: string | null
  sortOrder: number
  options: QuestionOptionWriteRequest[]
  version?: number
}

File: server/src/Elearning.Application/Exercises/Requests/QuestionWriteRequest.cs

using Elearning.Domain;

namespace Elearning.Application.Exercises;

public sealed record QuestionWriteRequest(
    string Text,
    QuestionType Type,
    string? Explanation,
    int SortOrder,
    IReadOnlyList<QuestionOptionWriteRequest> Options,
    QuestionPlacement Placement = QuestionPlacement.Reinforcement,
    int? VideoTimestampSeconds = null,
    long? Version = null);
